examples/showcase/src/main.rs

```diff
@@ -0,0 +1,279 @@
+use dashboard::Model as Dashboard;
+use fragments::Model as Fragments;
+use game_of_life::GameOfLife;
+use large_table::Model as LargeTable;
+use mount_point::Model as MountPoint;
+use npm_and_rest::Model as NpmAndRest;
+use npm_and_rest::gravatar::GravatarService;
+use npm_and_rest::ccxt::CcxtService;
+use textarea::Model as Textarea;
+use timer::Model as Timer;
+use todomvc::Model as Todomvc;
+use two_apps::Model as TwoApps;
+
+struct Context {
+    console: ConsoleService,
+    storage: StorageService,
+    dialog: DialogService,
+    web: FetchService,
+    ws: WebSocketService,
+    interval: IntervalService,
+    gravatar: GravatarService,
+    ccxt: CcxtService,
+    timeout: TimeoutService,
+    two_apps: two_apps::Context,
+}
+
+impl AsMut<ConsoleService> for Context {
+    fn as_mut(&mut self) -> &mut ConsoleService {
+        &mut self.console
+    }
+}
+
+impl AsMut<StorageService> for Context {
+    fn as_mut(&mut self) -> &mut StorageService {
+        &mut self.storage
+    }
+}
+
+impl AsMut<DialogService> for Context {
+    fn as_mut(&mut self) -> &mut DialogService {
+        &mut self.dialog
+    }
+}
+
+impl AsMut<FetchService> for Context {
+    fn as_mut(&mut self) -> &mut FetchService {
+        &mut self.web
+    }
+}
+
+impl AsMut<WebSocketService> for Context {
+    fn as_mut(&mut self) -> &mut WebSocketService {
+        &mut self.ws
+    }
+}
+
+impl AsMut<IntervalService> for Context {
+    fn as_mut(&mut self) -> &mut IntervalService {
+        &mut self.interval
+    }
+}
+
+impl custom_components::Printer for Context {
+    fn print(&mut self, data: &str) {
+        self.console.log(data);
+    }
+}
+
+impl AsMut<GravatarService> for Context {
+    fn as_mut(&mut self) -> &mut GravatarService {
+        &mut self.gravatar
+    }
+}
+
+impl AsMut<CcxtService> for Context {
+    fn as_mut(&mut self) -> &mut CcxtService {
+        &mut self.ccxt
+    }
+}
+
+impl AsMut<TimeoutService> for Context {
+    fn as_mut(&mut self) -> &mut TimeoutService {
+        &mut self.timeout
+    }
+}
+
+impl AsMut<two_apps::Context> for Context {
+    fn as_mut(&mut self) -> &mut two_apps::Context {
+        &mut self.two_apps
+    }
+}
+
+enum Scene {
+    NotSelected,
+    Counter,
+    Crm,
+    CustomComponents,
+    Dashboard,
+    Fragments,
+    GameOfLife,
+    LargeTable,
+    MountPoint,
+    NpmAndRest,
+    Textarea,
+    Timer,
+    Todomvc,
+    TwoApps,
+}
+
+enum Msg {
+    SwitchTo(Scene),
+}
+
+impl Component<Context> for Scene {
+    type Msg = Msg;
+    type Properties = ();
+
+    fn create(_: Self::Properties, _: &mut Env<Context, Self>) -> Self {
+        Scene::NotSelected
+    }
+
+    fn update(&mut self, msg: Self::Msg, _: &mut Env<Context, Self>) -> ShouldRender {
+        match msg {
+            Msg::SwitchTo(scene) => {
+                *self = scene;
+                true
+            }
+        }
+    }
+}
+
+impl Renderable<Context, Scene> for Scene {
```

Comment:
haha, this example _seriously_ needs a router :smile:

Reply:
Yeah, It's the preparation for the routing revolution ) But travis don't like a new building script. Some words about routing: I've been thinking about the routing a lot and I assume we could build it on traits with automatically routes constructing. I don't imagine how it will works but want to do some experiments soon.

Reply:
I'm honestly of the opinion that routing doesn't _need_ to be handled in `yew` directly at all -- instead make it a separate crate (at least while we are experimenting).

Reply:
In long-term I want to move all the services to separate crates. At present it is more convenient to keep everything together. I agree the router should be pluggable.